src/Servers/HttpSys/src/MessagePump.cs

```diff
@@ -275,4 +228,91 @@ public void Dispose()
 
         Listener.Dispose();
     }
+
+    private class AcceptLoop : IThreadPoolWorkItem
+    {
+        private readonly AsyncAcceptContext _asyncAcceptContext;
+        private readonly MessagePump _messagePump;
+        private readonly int _workerIndex;
+        private readonly bool _preferInlineScheduling;
+
+        public AcceptLoop(AsyncAcceptContext asyncAcceptContext,
+                          MessagePump messagePump,
+                          int workerIndex)
+        {
+            _asyncAcceptContext = asyncAcceptContext;
+            _messagePump = messagePump;
+            _workerIndex = workerIndex;
+            _preferInlineScheduling = _messagePump._options.UnsafePreferInlineScheduling;
```

Comment:
nit: Is there a reason for a field for this flag? Everything else is pulled off _messagePump

Reply:
Avoid multiple layers of indirection per request.

Reply:
Make it a local in `ExecuteAsync` above the `while`?

Reply:
That while loop lasts for a single iteration when things are being dispatched (the default scenario).